src/main/java/org/jabref/model/database/BibDatabaseContext.java

```diff
@@ -274,6 +276,77 @@ public Path getFulltextIndexPath() {
         return indexPath;
     }
 
+    /**
+     * Get the Git status of the database file.
+     *
+     * @return The Git status if the database is under version control, or empty if not.
+     */
+    public Optional<GitHandler.GitStatus> getGitStatus() {
+        if (!isUnderVersionControl() || getDatabasePath().isEmpty()) {
+            // if the file is not under version control or does not have a datapath return empty
+            return Optional.empty();
+        }
+
+        try {
+            // get the path of the current file, then set false to not create a repo by default
+            GitHandler gitHandler = new GitHandler(getDatabasePath().get(), false);
+            return gitHandler.getFileStatus(getDatabasePath().get());
+        } catch (Exception e) {
+            // Silent fail on Git status errors
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Returns whether this database is under version control (e.g., git).
+     */
+    public boolean isUnderVersionControl() {
+        if (underVersionControl != null) {
+            return underVersionControl;
+        }
+
+        if (getDatabasePath().isEmpty()) {
+            underVersionControl = false;
```

Comment:
autoboxing is expensive. If underVersionControl needs to be a Boolean, you should use Boolean.FALSE here.